supervision/detection/utils.py

```diff
@@ -1039,3 +1039,60 @@ def cross_product(anchors: np.ndarray, vector: Vector) -> np.ndarray:
     )
     vector_start = np.array([vector.start.x, vector.start.y])
     return np.cross(vector_at_zero, anchors - vector_start)
+
+
+def get_box_intersection(
+    xyxy_1: np.ndarray, xyxy_2: np.ndarray
+) -> Optional[np.ndarray]:
+    overlap_xmin = max(xyxy_1[0], xyxy_2[0])
+    overlap_ymin = max(xyxy_1[1], xyxy_2[1])
+    overlap_xmax = min(xyxy_1[2], xyxy_2[2])
+    overlap_ymax = min(xyxy_1[3], xyxy_2[3])
+
+    if overlap_xmin < overlap_xmax and overlap_ymin < overlap_ymax:
+        return np.array([overlap_xmin, overlap_ymin, overlap_xmax, overlap_ymax])
+    else:
+        return None
+
+
+def get_unit_vector(xy_1: np.ndarray, xy_2: np.ndarray) -> np.ndarray:
+    direction = xy_2 - xy_1
+    magnitude = np.linalg.norm(direction)
+    return direction / magnitude if magnitude > 0 else np.zeros(2)
+
+
+def spread_out_boxes(
+    xyxy: np.ndarray, step: int, max_iterations: int = 100
+) -> np.ndarray:
+    if len(xyxy) == 0:
+        return xyxy
+
+    xyxy_padded = pad_boxes(xyxy, px=step)
+    for _ in range(max_iterations):
+        iou = box_iou_batch(xyxy_padded, xyxy_padded)
+        np.fill_diagonal(iou, 0)
+
+        if np.all(iou == 0):
+            break
+
+        i, j = np.unravel_index(np.argmax(iou), iou.shape)
```

Comment:
Why are we processing only one box pair (the one with the greatest overlap) per iteration? It seems like it should be possible to process all overlapping boxes in parallel.